File: app/src/main/java/com/my/televip/virtuals/messenger/ImageReceiver.java
package com.my.televip.virtuals.messenger;

import com.my.televip.obfuscate.ArgsResolver;
import com.my.televip.obfuscate.Obfuscate;

import de.robv.android.xposed.XposedHelpers;

public class ImageReceiver {

    public Object imageReceiver;

    public ImageReceiver(Object imageReceiver){
        this.imageReceiver = imageReceiver;
    }

    public ImageLocation getImageLocation() {
        return new ImageLocation(XposedHelpers.callMethod(imageReceiver, Obfuscate.getMethodName("ImageReceiver", "getImageLocation")));
    }

}

File: app/src/main/java/com/my/televip/virtuals/messenger/ImageReceiver.kt
package com.my.televip.virtuals.messenger

import com.my.televip.obfuscate.AutomationResolver
import de.robv.android.xposed.XposedHelpers

class ImageReceiver(@JvmField val imageReceiver: Any?) {
    fun getImageLocation(): ImageLocation {
        return ImageLocation(
            XposedHelpers.callMethod(
                imageReceiver,
                AutomationResolver.resolve("ImageReceiver", "getImageLocation", AutomationResolver.ResolverType.Method)
            )
        )
    }
}
